GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCOREvent+GDTCCTSupport.m

```diff
@@ -0,0 +1,194 @@
+
+#import <GoogleDataTransport/GDTCORConsoleLogger.h>
+
+NSString *const GDTCCTNeedsNetworkConnectionInfo = @"needs_network_connection_info";
+
+NSString *const GDTCCTNetworkConnectionInfo = @"network_connection_info";
+
+NSString *const GDTCCTEventCodeInfo = @"event_code_info";
+
+@implementation GDTCOREvent (GDTCCTSupport)
+
+- (void)setNeedsNetworkConnectionInfoPopulated:(BOOL)needsNetworkConnectionInfoPopulated {
+  if (!needsNetworkConnectionInfoPopulated) {
+    self.customBytes = nil;
+  } else {
+    @try {
+      NSError *error;
+      NSMutableDictionary *bytesDict;
+      if (self.customBytes) {
+        bytesDict = [[NSJSONSerialization JSONObjectWithData:self.customBytes
+                                                     options:0
+                                                       error:&error] mutableCopy];
+        if (error) {
+          GDTCORLogDebug(@"Error when setting an even'ts event_code: %@", error);
+          return;
+        }
+      } else {
+        bytesDict = [[NSMutableDictionary alloc] init];
+      }
+      [bytesDict setObject:@YES forKey:GDTCCTNeedsNetworkConnectionInfo];
+      self.customBytes =
+      [NSJSONSerialization dataWithJSONObject:bytesDict
+                                      options:0
+                                        error:&error];
+    } @catch (NSException *exception) {
+      GDTCORLogDebug(@"Error when setting the event for needs_network_connection_info: %@",
+                     exception);
+    }
+  }
+}
+
+- (BOOL)needsNetworkConnectionInfoPopulated {
+  if (self.customBytes) {
+    @try {
+      NSError *error;
+      NSDictionary *bytesDict = [NSJSONSerialization JSONObjectWithData:self.customBytes
+                                                                options:0
+                                                                  error:&error];
+      return bytesDict && !error && [bytesDict[GDTCCTNeedsNetworkConnectionInfo] boolValue];
+    } @catch (NSException *exception) {
+      GDTCORLogDebug(@"Error when checking the event for needs_network_connection_info: %@",
+                     exception);
+    }
+  }
+  return NO;
+}
+
+- (void)setNetworkConnectionInfoData:(NSData *)networkConnectionInfoData {
+  @try {
+    NSError *error;
+    NSString *dataString = [networkConnectionInfoData base64EncodedStringWithOptions:0];
+    if (dataString) {
+      NSMutableDictionary *bytesDict;
+      if (self.customBytes) {
+        bytesDict = [[NSJSONSerialization JSONObjectWithData:self.customBytes
+                                                     options:0
+                                                       error:&error] mutableCopy];
+        if (error) {
+          GDTCORLogDebug(@"Error when setting an even'ts event_code: %@", error);
+          return;
+        }
+      } else {
+        bytesDict = [[NSMutableDictionary alloc] init];
+      }
+      [bytesDict setObject:dataString forKey:GDTCCTNetworkConnectionInfo];
```

Comment:
Similarly, check dataString for nil

Reply:
This is already surrounded by an if (dataString != nil)